wrapper/src/main/java/software/amazon/jdbc/util/events/PeriodicEventPublisher.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.util.events;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import software.amazon.jdbc.util.StringUtils;
+
+/**
+ * An event publisher that periodically sends out all messages received during the latest interval of time. Messages are
+ * recorded in a set so that messages of equivalent value are not duplicated in the same message batch.
+ */
+public class PeriodicEventPublisher implements EventPublisher {
+  protected static final long DEFAULT_MESSAGE_INTERVAL_NANOS = TimeUnit.SECONDS.toNanos(30);
+  protected final Map<Class<? extends Event>, Set<EventSubscriber>> subscribers = new ConcurrentHashMap<>();
```

Comment:
Should we use weak references to EventSubscriber?

Reply:
Hmm good question. What is the problem we would be trying to avoid? I know weak references help with garbage collection but am unsure which scenario would cause a garbage collection problem in this case